src/dmd/s2ir.d

```diff
@@ -754,6 +766,16 @@ private extern (C++) class S2irVisitor : Visitor
                                 writetohp = true;
                             }
                         }
+                        Type t = ce.e1.type.toBasetype();
+                        if (t.ty == Tdelegate)
+                            t = t.nextOf();
+                        if (t.ty == Tfunction && retStyle(cast(TypeFunction)t) == RET.stack)
+                        {
+                            irs.ehidden = el_var(irs.shidden);
+                            e = toElemDtor(s.exp, irs);
+                            e = el_una(OPaddr, TYnptr, e);
+                            goto L1;
+                        }
```

Comment:
These code blocks look identical.  Can they be combined into a reusable entity somehow?

Reply:
I thought I'd leave refactoring for another PR, especially if there turns out to be a problem with this code.